pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -1630,3 +1631,208 @@ func TestDeleteOfBareMetalHostDeletesMachine(t *testing.T) {
+
+	scheme := runtime.NewScheme()
+	clusterapis.AddToScheme(scheme)
+	bmoapis.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	c := fakeclient.NewFakeClientWithScheme(scheme)
+
+	actuator, err := NewActuator(ActuatorParams{
+		Client: c,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	c.Create(context.TODO(), machine)
+	c.Create(context.TODO(), host)
+	c.Create(context.TODO(), node)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	host = &bmh.BareMetalHost{}
+	c.Get(context.TODO(), hostNamespacedName, host)
+	if !hasPowerOffRequestAnnotation(host) {
+		t.Log("Expected reboot annotation on the host but none found")
+		t.Fail()
+	}
+
+	host.Status.PoweredOn = false
+	c.Update(context.TODO(), host)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	machine = &machinev1.Machine{}
+	c.Get(context.TODO(), machineNamespacedName, machine)
+
+	if _, exists := machine.Annotations[poweredOffForRemediation]; !exists {
+		t.Log("Expected powered-off-for-remediation annotation to exist on machine but none found")
+		t.Fail()
+	}
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		switch err.(type) {
+			case *clustererror.RequeueAfterError:
+				break
+			default:
+				t.Errorf("unexpected error %v", err)
+		}
+	}
+
+	node = &corev1.Node{}
+	err = c.Get(context.TODO(), nodeNamespacedName, node)
+	if !errors.IsNotFound(err) {
+		t.Log("Expected node to be deleted")
+		t.Fail()
+	}
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	host = &bmh.BareMetalHost{}
+	c.Get(context.TODO(), hostNamespacedName, host)
+
+	if hasPowerOffRequestAnnotation(host) {
+		t.Log("Expected reboot annotation to be removed but it's still there")
+		t.Fail()
+	}
+
+	//simulate power on and node that comes up again
+	host.Status.PoweredOn = true
+	c.Update(context.TODO(), host)
+
+	node, _ = getNode("node1")
+	linkMachineAndNode(machine, node)
+	c.Create(context.TODO(), node)
+	c.Update(context.TODO(), machine)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	machine = &machinev1.Machine{}
+	c.Get(context.TODO(), machineNamespacedName, machine)
+
+	if len(machine.Annotations) > 0 {
+		if _, exists := machine.Annotations[externalRemediationAnnotation]; exists {
+			t.Log("Expected external remediation annotation to be removed but it's still there")
+			t.Fail()
+		}
+
+		if _, exists := machine.Annotations[poweredOffForRemediation]; exists {
+			t.Log("Expected powered-off-for-remediation annotation to be removed but it's still there")
+			t.Fail()
+		}
+	}
+
+	//make sure nothing happens after remediation
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+	err = actuator.Update(context.TODO(), nil, machine)
```

Comment:
Is there something special about calling Update() twice instead of once or three times?

Reply:
Not something specific I can think of. I can remove the second one if you want me to. 
Just wanted to make sure that another power off request is not commencing, even after multiple calls to `Update()`

Reply:
Well, we call it twice but don't do anything in between so it wasn't clear why. It's probably not important.